bind-mount.c

```diff
@@ -455,10 +455,71 @@ bind_mount (int           proc_fd,
               /* If we can't read the mountpoint we can't remount it, but that should
                  be safe to ignore because its not something the user can access. */
               if (errno != EACCES)
-                return 5;
+                return BIND_MOUNT_ERROR_REMOUNT_SUBMOUNT;
             }
         }
     }
 
-  return 0;
+  return BIND_MOUNT_SUCCESS;
+}
+
+void
+die_with_bind_result (bind_mount_result res,
+                      int               saved_errno,
+                      const char       *format,
+                      ...)
+{
+  va_list args;
+  bool want_errno = TRUE;
+
+  fprintf (stderr, "bwrap: ");
+
+  va_start (args, format);
+  vfprintf (stderr, format, args);
+  va_end (args);
+
+  fprintf (stderr, ": ");
+
+  switch (res)
+    {
+      case BIND_MOUNT_ERROR_MOUNT:
+        fprintf (stderr, "Unable to mount source on destination");
```

Comment:
I think this would be more elegant with a helper method equivalent of `strerror()` - i.e. it returns a `const char *` instead of doing the printing itself.  (Or analogus to `impl Display for BindError` in Rust).

Reply:
One problem with doing this is that if we get an unknown/invalid `bind_mount_error` , we can't fall back to printing it with `%d`, unless the helper method returns a newly-allocated `char *` (or a static buffer, I suppose, but eww).

Reply:
New version adds `bind_mount_result_to_string()`, does that look better to you?